source/dlangbot/database.d

```diff
@@ -0,0 +1,109 @@
+/**
+Implements a database to keep track of the Points scored
+by Authors that fix bugs. The table has the following
+columns:
+
+Github account (Primary key)
+Total Points
+*/
+module dlangbot.database;
+
+import ae.sys.database : Database;
+
+private Database db;
+private string[] severities = ["enhancement", "trivial", "minor",
+            "normal", "major", "critical", "blocker", "regression"];
+enum table_schema = q"SQL
+CREATE TABLE [LeaderBoard] (
+[Id]  INTEGER PRIMARY KEY NOT NULL,
+[Name] TEXT NOT NULL,
+[Enhancement] INTEGER,
+[Trivial] INTEGER,
+[Minor] INTEGER,
+[Normal] INTEGER,
+[Major] INTEGER,
+[Critical] INTEGER,
+[Blocker] INTEGER,
+[Regression] INTEGER,
+[Points] INTEGER
+);
+SQL";
+
+/**
+Checks whether a database file already exists.
+If that is the case, it returns the database,
+otherwise it creates it.
+*/
+Database initDatabase(string databasePath)
+{
+    import std.file : exists;
+    import ae.sys.file : ensurePathExists;
+
+    version(unittest)
+    {
+        if (databasePath == ":memory:")
+            return Database(databasePath, [table_schema,]);
+    }
+
+    if (databasePath.exists)
+        return Database(databasePath);
```

Comment:
You need to pass the schema to the database constructor always, otherwise successive additions (migration instructions) won't get executed.